data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchSourceConfiguration.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source.opensearch;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.AwsAuthenticationConfiguration;
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.ConnectionConfiguration;
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.IndexParametersConfiguration;
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.QueryParameterConfiguration;
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.SchedulingParameterConfiguration;
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.RetryConfiguration;
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.SearchConfiguration;
+
+
+import java.util.Map;
+
+public class OpenSearchSourceConfiguration {
+
+    @JsonProperty("connection")
+    private ConnectionConfiguration connectionConfiguration;
+
+    @JsonProperty("indices")
+    private IndexParametersConfiguration indexParametersConfiguration;
+
+    @JsonProperty("aws")
+    private AwsAuthenticationConfiguration awsAuthenticationOptions;
+
+    @JsonProperty("scheduling")
+    private SchedulingParameterConfiguration schedulingParameterConfiguration;
+
+    @JsonProperty("query")
+    private QueryParameterConfiguration queryParameterConfiguration;
+
+    @JsonProperty("search_options")
+    private SearchConfiguration searchConfiguration;
+
+    @JsonProperty("retry")
+    private RetryConfiguration retryConfiguration;
+
+    private Map<String,String> indexNames;
```

Comment:
This map will not be needed. Please remove

Reply:
Thanks @graytaylor0  for the review comments. The changes are incorporated.